### PR TITLE
[arm] fix stack alignment for structs

### PR DESCRIPTION
**Pull request by lewurm**
fixes https://github.com/mono/mono/issues/7449

/cc @joncham

### CODE REVIEW

**Comment by vargaz** [comment]
Looks good in theory, hopefully it won't break anything.

**Comment by lewurm** [comment]
I've tested it on Linux armv5 and armv7. I'm a bit concerned about iOS 32bit.

File: mono/tests/pinvoke11.cs [hunk 1]
```diff
@@ -159,6 +196,12 @@ static int Main()
 		if (f.fi.f1 != 1.0)
 			return 12;
 
+		Rectangle rect = new Rectangle (10, 10, 100, 20);
+		Scalar4 sc4 = new Scalar4 (32, 64, 128, 256);
+		var sc4_ret = mono_return_struct_4_double (IntPtr.Zero, rect, sc4, 0x1337, 0x1234, 0x9876);
+		if (sc4_ret != "sc4 = {32.0, 64.0, 128.0, 256.0 }, a=1337, b=1234, c=9876\n")
+			return 13;
```
Comment (luhenry) [inline mono/tests/pinvoke11.cs]: You should print `sc4_ret` in case it fails, so we have some information in CI.

**Comment by luhenry** [comment]
@monojenkins build Linux ARMv7

**Pull request merged**